### PR TITLE
[CARBONDATA-2708][BloomDataMap]  clear index file in case of data load failure

### PR DESCRIPTION
**Pull request by xuchuanyin**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by xuchuanyin** [comment]
This PR depends on PR #2461

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6943/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5726/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6954/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5716/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6966/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5746/

**Comment by xuchuanyin** [comment]
retest sdv please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5750/

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 1]
```diff
@@ -588,6 +591,21 @@ object CarbonDataRDDFactory {
...
   }
 
+  /**
+   * clear datamap files for segment
+   */
+  private def clearDataMapFiles(carbonTable: CarbonTable, segmentId: String): Unit = {
+    try {
+      DataMapStoreManager.getInstance().getAllDataMap(carbonTable).asScala
+        .filter(_.getDataMapSchema.isIndexDataMap).foreach { p =>
+        p.deleteDatamapData(List(new Segment(segmentId)).asJava)
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: move `List(new Segment(segmentId)).asJava` before the loop and move `foreach` to next line, to make it better formatted

File: datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapFactoryBase.java [hunk 2]
```diff
@@ -270,6 +260,22 @@ public void clear() {
 
   }
 
+  @Override public void deleteDatamapData(Segment segment) {
```
Comment (jackylk) [inline datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapFactoryBase.java]: move @Override to previous line

File: datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapFactoryBase.java [hunk 3]
```diff
@@ -270,6 +260,22 @@ public void clear() {
...
+      String segmentId = segment.getSegmentNo();
+      String datamapPath = CarbonTablePath
+          .getDataMapStorePath(tableIdentifier.getTablePath(), segmentId, dataMapName);
+      if (FileFactory.isFileExist(datamapPath)) {
+        CarbonFile file = FileFactory.getCarbonFile(datamapPath,
+            FileFactory.getFileType(datamapPath));
+        CarbonUtil.deleteFoldersAndFilesSilent(file);
+      }
+    } catch (IOException | InterruptedException ex) {
+      throw new RuntimeException(
```
Comment (jackylk) [inline datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapFactoryBase.java]: Seems throw IOException is better, then catch it and log it in out most caller

Comment (jackylk) [reply datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapFactoryBase.java]: please do the same for line 279

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala (hunk 1 above)
Comment (xuchuanyin) [reply integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: OK

File: datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapFactoryBase.java (hunk 2 above)
Comment (xuchuanyin) [reply datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapFactoryBase.java]: OK

File: datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapFactoryBase.java (hunk 3 above)
Comment (xuchuanyin) [reply datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapFactoryBase.java]: OK

**Comment by xuchuanyin** [comment]
review comments are resolved

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7022/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5801/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5773/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7041/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5818/

**Comment by xuchuanyin** [comment]
retest sdv please

**Comment by jackylk** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7095/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5871/

**Pull request closed without merging**